src/views/preview/project-view.jsx

```diff
@@ -103,11 +106,20 @@ class Preview extends React.Component {
             (this.state.projectId !== prevState.projectId))) {
             this.fetchCommunityData();
             this.getProjectData(this.state.projectId);
-            this.setState({justShared: false}); // eslint-disable-line react/no-did-update-set-state
+            if (this.state.justShared) {
+                this.setState({ // eslint-disable-line react/no-did-update-set-state
+                    justShared: false
+                });
+            }
         }
         if (this.state.projectId === '0' && this.state.projectId !== prevState.projectId) {
             this.props.resetProject();
-            this.setState({justShared: false}); // eslint-disable-line react/no-did-update-set-state
+            if (this.state.justRemixed || this.state.justShared) {
```

Comment:
is this `if` needed, or should it always reset these states (like the project)

Reply:
Unfortunately, `setState` will cause a re-render even if the values you set it to are identical to the current values. So this just avoids an unnecessary re-render, but will always re-render if `justRemixed` or `justShared` were true.

see https://itnext.io/react-setstate-usage-and-gotchas-ac10b4e03d60
![image](https://user-images.githubusercontent.com/3431616/49607846-24d8c800-f965-11e8-93ce-e1e7ad17b7ec.png)